### DISCUSSION TITLE
Help needed

### DISCUSSION THREAD

**Discussion by oliverzh** [OpenVPN/openvpn3#299] [Q&A]
Hi,

I'm new to C#. I'm trying to build a VPN client for windows using openvpn3. 

I have followed the steps for windows and cmake into the build folder.
> git clone https://github.com/OpenVPN/openvpn3.git core && cd core
> cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=<path_to_vcpkg>\scripts\buildsystems\vcpkg.cmake -DVCPKG_OVERLAY_PORTS=deps\vcpkg-ports
> cmake --build build --config Release --target ovpncli

I assume next steps are to 
1. include the correct .hpp files into my winform project 
2. use the correct class to create the vpn connection using my .ovpn file

Have I got the steps right? where can I find some documentation for doing this?

**Comment by OPstriker** [comment]
I'm not so familiar with C#, but you need to wrap the openvpn3 library to work with C#. I'm pretty sure you can do it with swig which is provided in the repo.
